### ISSUE TITLE
Race condition during elemental-system-agent to rancherd-system-agent handoff prevents cluster provisioning to start

### ISSUE THREAD

**Issue by fgiudici** [rancher/elemental-operator#188]
Environment:
- on cloud Rancher
- on premises machine to provision

The OS provisioning on the new machine works fine: after reboot the machine is never ready to be cluster provisioned.
MachineInventory Status reports: _waiting for plan to be applied_.

The elemental-system-agent is found stopped and the rancher-system-agent is already running.
Stopping for a while  the rancher-system-agent and restarting the elemental-system-agent unblocks the cluster provisioning.

**Comment by fgiudici** [comment]
duplicate of https://github.com/rancher/elemental/issues/374
